swift/example_code/s3/presigned-urls/Sources/presigned-upload/entry.swift

```diff
@@ -0,0 +1,221 @@
+        }
+        print("Uploaded \(sourcePath) to \(bucket)/\(fileName).")
+    }
+    // snippet-end:[swift.s3.presigned.upload-file]
+
+    // -MARK: - Presigned upload
+
+    // snippet-start:[swift.s3.presigned.presign-upload-file]
+    /// Upload the specified file to the Amazon S3 bucket with the given name.
+    /// The request is presigned.
+    ///
+    /// - Parameters:
+    ///   - sourcePath: The pathname of the source file on a local disk.
+    ///   - bucket: The Amazon S3 bucket to store the file into.
+    ///   - key: The key (name) to give the uploaded file. The filename of the
+    ///     source is used by default.
+    ///
+    /// - Throws: `TransferError.uploadError`
+    func uploadFilePresigned(sourcePath: String, bucket: String, key: String?) async throws {
+        let fileURL = URL(fileURLWithPath: sourcePath)
+        let fileName: String
+        
+        // If no key was provided, use the last component of the filename.
+        
+        if key == nil {
+            fileName = fileURL.lastPathComponent
+        } else {
+            fileName = key!
+        }
+        
+        // Create an Amazon S3 client in the desired Region.
+
+        let config = try await S3Client.S3ClientConfiguration(region: region)
+        let s3Client = S3Client(config: config)
+        
+        // Look to see if the file already exists in the target bucket.
+        
+        do {
+            let headInput = HeadObjectInput(bucket: bucket, key: fileName)
+            let headResult = try await s3Client.headObject(input: headInput)
+            
+            // If HeadObject is successful, the file definitely exists.
+            
+            print("File exists with ETag \(headResult.eTag ?? "<no ETag>"). Skipping upload.")
+            return
+        } catch let error as AWSServiceError {
+            switch(error.errorCode) {
+            case "NotFound":
+                break
+            default:
+                throw TransferError.readError
+            }
+        }
+
+        print("Uploading file from \(fileURL.path) to \(bucket)/\(fileName).")
+
+        // Presign the `PutObject` request with a 10-minute expiration. The
+        // presigned request has a custom configuration that asks for up to
+        // six attempts to put the file.
+        //
+        // The presigned `URLRequest` can then be sent using the URL Loading System
+        // (https://developer.apple.com/documentation/foundation/url_loading_system/uploading_data_to_a_website).
+
+        // snippet-start:[swift.s3.presigned.presign-PutObject-advanced]
+        let fileData = try Data(contentsOf: fileURL)
+        let dataStream = ByteStream.data(fileData)
+        let presignedURL: URL
+
+        let putConfig = try await S3Client.S3ClientConfiguration(
+            maxAttempts: 6,
+            region: region
+        )
+
+        config.maxAttempts = 6
```

Comment:
This looks redundant. maxAttempts was set in the initializer.

Reply:
Fixed, and moved the comment explaining things up above the configuration to make sense.